doc/source/rllib-algorithms.rst

```diff
@@ -608,3 +560,77 @@ Tuned examples: `SimpleContextualBandit <https://github.com/ray-project/ray/blob
+
+**MADDPG-specific configs** (see also `common configs <rllib-training.html#common-parameters>`__):
+
+Tuned examples: `Multi-Agent Particle Environment <https://github.com/wsjeon/maddpg-rllib/tree/master/plots>`__, `Two-step game <https://github.com/ray-project/ray/blob/master/rllib/examples/two_step_game.py>`__
+
+.. literalinclude:: ../../rllib/contrib/maddpg/maddpg.py
+   :language: python
+   :start-after: __sphinx_doc_begin__
+   :end-before: __sphinx_doc_end__
+
+.. _alphazero:
+
+Single-Player Alpha Zero (contrib/AlphaZero)
+--------------------------------------------
+|pytorch|
+`[paper] <https://arxiv.org/abs/1712.01815>`__ `[implementation] <https://github.com/ray-project/ray/blob/master/rllib/contrib/alpha_zero>`__ AlphaZero is an RL agent originally designed for two-player games. This version adapts it to handle single player games. The code can be sscaled to any number of workers. It also implements the ranked rewards `(R2) <https://arxiv.org/abs/1807.01672>`__ strategy to enable self-play even in the one-player setting. The code is mainly purposed to be used for combinatorial optimization.
+
+Tuned examples: `CartPole-v0 <https://github.com/ray-project/ray/blob/master/rllib/contrib/alpha_zero/examples/train_cartpole.py>`__
+
+**AlphaZero-specific configs** (see also `common configs <rllib-training.html#common-parameters>`__):
+
+.. literalinclude:: ../../rllib/contrib/alpha_zero/core/alpha_zero_trainer.py
+   :language: python
+   :start-after: __sphinx_doc_begin__
+
+Parameter Sharing
+-----------------
+
+Parameter Sharing `[paper] <http://ala2017.it.nuigalway.ie/papers/ALA2017_Gupta.pdf>` and `[paper] <https://arxiv.org/abs/2005.13625>` `[implementation] <github.com/parametersharingmadrl/parametersharingmadrl>` in multi-agent RL is a class of methods that take a base single agent method, and use it to learn a single policy for all agents. As naive as this seems, this is shown to achieve state of the art performance in cooperative games, and parameter sharing good DRL methods is usually how you should start trying to learn a multi-agent problem. An example implementation of this in RLlib is included in the second paper. Additionally, while this seems like can only work for environments with homogeneous agents, this is not the case. The second paper introduces how to do this for every case of heterogeneity, and simple wrappers that apply the methods to [PettingZoo](https://github.com/PettingZoo-Team/PettingZoo) API environments are available in the [SuperSuit](https://github.com/PettingZoo-Team/SuperSuit) package.
+
+Fully Independent Learning
+--------------------------
+Fully Independent Learning `[implementation] <github.com/parametersharingmadrl/parametersharingmadrl>` is when you have a collection of agents learning independently of each other via single agent methods. This typically works, but is inferior to dedicated multi-agent RL methods. 
```

Comment:
Can you explain how to implement this in RLlib?

Reply:
Again, covered in implementation link